### PR TITLE
Base events on logical changes (not webhooks)

### PR DESCRIPTION
**Pull request by avnerbraverman**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,26 +1,65 @@
...
+  ChangeColumnValue: 'change_column_value',
+  ChangeSpecificColumnValue: 'change_specific_column_value',
+  CreateItem: 'create_item',
+  CreateUpdate: 'create_update',
+}
+
+const SepytTneve = {
+  incoming_notification: 'IncomingNotification',
+  change_column_value: 'ChangeColumnValue',
+  update_column_value: 'ChangeColumnValue',
```
Comment (lagache) [inline src/index.ts:36]: ChangeColumnValue => UpdateColumnValue

File: src/index.ts [hunk 2]
```diff
@@ -1,26 +1,65 @@
...
+  change_specific_column_value: 'ChangeSpecificColumnValue',
+  update_specific_column_value: 'ChangeSpecificColumnValue',
```
Comment (lagache) [inline src/index.ts:38]: ChangeSpecificColumnValue => UpdateSpecificColumnValue

File: src/index.ts [hunk 3]
```diff
@@ -1,26 +1,65 @@
...
+  create_item: 'CreateItem',
+  create_update: 'CreateUpdate',
+}
+
+interface MondayEvent {
+  userId: string
+  originalTriggerUuid: any
+  boardId: string
+  groupId: string
+  itemId: string // pulseId
```
Comment (lagache) [inline src/index.ts:48]: wrong comment?

File: src/index.ts [hunk 4]
```diff
@@ -1,26 +1,65 @@
...
+  pulseId: string // pulseId
+  itemName: string // pulseName
```
Comment (lagache) [inline src/index.ts:50]: wrong comment?

File: src/index.ts [hunk 5]
```diff
@@ -365,34 +433,38 @@ export default class MondayConnector extends BaseHttpConnector<
...
-      config,
-    })
+    columnId?: string,
+  ): Promise<string> {
+    const params: any = { board_id: Number(boardId), url, event }
+    if (event === 'ChangeSpecificColumnValue') {
+      params.config = { columnId }
+    }
+    const res = await this.query(CREATE_WEBHOOK_QUERY, params)
+    return (res as any).create_webhook.id
```
Comment (gabeeden) [inline src/index.ts:447]:
```suggestion
    return res?.create_webhook.id
```

File: src/index.ts [hunk 6]
```diff
@@ -138,72 +175,103 @@ export default class MondayConnector extends BaseHttpConnector<
...
   _sdk: MondaySdk
+  private webhookURL: string
+  private webhookId?: string
+  private webhookLastChangedAt?: number
 
   constructor(app: Reshuffle, options: MondayConnectorConfigOptions, id?: string) {
     super(app, options, id)
+    this.webhookURL = validateBaseURL(options.baseURL) + WEBHOOK_PATH
     this._sdk = mondaySdk({ token: options.token })
+    this.app.registerHTTPDelegate(WEBHOOK_PATH, this)
```
Comment (lagache) [inline src/index.ts:186]: I'm tempted to move this in the on function. You may want to use only the actions and so don't need to register a webhook.

File: src/index.ts [hunk 7]
```diff
@@ -365,34 +433,38 @@ export default class MondayConnector extends BaseHttpConnector<
...
-    config?: Record<string, unknown>,
-  ): Promise<MondayApiReponse['data']> {
-    return this.query(CREATE_WEBHOOK_QUERY, {
-      board_id: boardId,
-      url,
-      event,
-      config,
-    })
+    columnId?: string,
+  ): Promise<string> {
```
Comment (gabeeden) [inline src/index.ts:441]:
```suggestion
  ): Promise<string | undefined> {
```

File: src/index.ts [hunk 8]
```diff
@@ -1,26 +1,65 @@
...
+
+const EventTypes = {
+  IncomingNotification: 'incoming_notification',
+  ChangeColumnValue: 'change_column_value',
+  ChangeSpecificColumnValue: 'change_specific_column_value',
+  CreateItem: 'create_item',
+  CreateUpdate: 'create_update',
+}
+
+const SepytTneve = {
```
Comment (gabeeden) [inline src/index.ts:33]: Confused what the name means here

**Review by gabeeden — APPROVED** [review]

File: src/index.ts [hunk 9]
```diff
@@ -138,72 +175,103 @@ export default class MondayConnector extends BaseHttpConnector<
...
+    ) {
+      throw new Error(`Invalid column id for type ChangeSpecificColumnValue: ${options.columnId}`)
     }
-    const event = new EventConfiguration(eventId, this, options)
-    event.options.path = event.options.path || DEFAULT_WEBHOOK_PATH
-    event.options.deleteWebhookOnExit = event.options.deleteWebhookOnExit || true
-    this.eventConfigurations[event.id] = event
 
+    const event = new EventConfiguration(
+      eventId || `Monday${options.boardId}.${options.type}/${this.id}`,
```
Comment (lagache) [inline src/index.ts:209]: I'd use Monday/${options.type}/${options.boardId}/${this.id} as it's easier to read and more consistent with other connectors

File: src/index.ts (hunk 5 above)
Comment (avnerbraverman) [reply src/index.ts:447]: Changed to res! so that we don't fail silently

File: src/index.ts (hunk 7 above)
Comment (avnerbraverman) [reply src/index.ts:441]: No need (see^)

File: src/index.ts [hunk 10]
```diff
@@ -138,72 +175,103 @@ export default class MondayConnector extends BaseHttpConnector<
...
+      this,
+      { boardId, type: options.type, columnId: options.columnId },
```
Comment (lagache) [inline src/index.ts:211]: i think options here would do the trick.
e.g. new EventConfiguration(..., this, options)

File: src/index.ts (hunk 8 above)
Comment (avnerbraverman) [reply src/index.ts:33]: Its' "EventTypes" in reverse. I'll change

File: src/index.ts (hunk 5 above)
Comment (avnerbraverman) [reply src/index.ts:447]: Your lint doesn't like "!"s. tested explicitly

File: src/index.ts (hunk 8 above)
Comment (lagache) [reply src/index.ts]: haha I was wondering too: brie...lliant

File: src/index.ts [hunk 11]
```diff
@@ -365,34 +433,41 @@ export default class MondayConnector extends BaseHttpConnector<
...
+      throw new Error(`Failed to create webhook: boardId=${boardId} event=${event}`)
+    }
+    return res.create_webhook.id
   }
 
   deleteWebhook(id: number): Promise<MondayApiReponse['data']> {
     return this.query(DELETE_WEBHOOK_QUERY, { id })
   }
 
-  onRemoveEvent(event: EventConfiguration): void {
```
Comment (lagache) [inline src/index.ts:386]: don't we need that method? where do we call `deleteWebhook`?

File: src/index.ts (hunk 2 above)
Comment (avnerbraverman) [reply src/index.ts:38]: Clarified in comments

File: src/index.ts (hunk 1 above)
Comment (avnerbraverman) [reply src/index.ts:36]: Clarified in comments

File: src/index.ts (hunk 3 above)
Comment (avnerbraverman) [reply src/index.ts:48]: Clarified in comments

File: src/index.ts (hunk 4 above)
Comment (avnerbraverman) [reply src/index.ts:50]: Clarified in comments

File: src/index.ts (hunk 9 above)
Comment (avnerbraverman) [reply src/index.ts:209]: Fixed

File: src/index.ts (hunk 10 above)
Comment (avnerbraverman) [reply src/index.ts:211]: Yes, but I wanted to explicitly copy in case someone passed some invalid fields in

File: src/index.ts (hunk 11 above)
Comment (avnerbraverman) [reply src/index.ts:386]: Right! added back

**Pull request merged**